### PR TITLE
Feature/xhr migration

### PR DESCRIPTION
**Pull request by yusufkhan07**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [X] Build (`npm run build`) was run locally and any changes were pushed

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: 58

## What is the new behavior?

- Nanogram now users XHR instead of fetch API. Any polyfills used for Fetch API can be removed.
- Nodejs doesn't support XHR, users you can use any library with a compatible interface.

## Does this introduce a breaking change?

- [X] Yes
- [ ] No

This commit brings a breaking change to nodejs users. We've migrated from Fetch API to XHR. Node users need to use a library with a compatible interface, such as [xmlhttprequest](https://www.npmjs.com/package/xmlhttprequest)

### CODE REVIEW

File: src/nanogram.ts [hunk 1]
```diff
@@ -49,15 +49,21 @@ export default class Nanogram {
...
+        xhr.onload = (): void => resolve(xhr);
+        xhr.onerror = reject;
+        xhr.send();
+      });
     };
 
-    const response = await fetch(request, requestOptions).then((response: Response) => {
-      if (response.ok) {
-        return response.text();
+    // redirect: 'follow',
```
Comment (webistomin) [inline src/nanogram.ts:63]: Forgot to delete?

File: src/nanogram.ts [hunk 2]
```diff
@@ -49,15 +49,21 @@ export default class Nanogram {
   /*
    * Make fetch call to instagram
    */
-  private async HTTP<T>(request: RequestInfo, useRegExp = true): Promise<T | undefined> {
-    const requestOptions: RequestInit = {
-      method: 'GET',
-      redirect: 'follow',
+  private async HTTP<T>(request: string, useRegExp = true): Promise<T | undefined> {
+    const xhrrequest = (method: string, url: string): Promise<XMLHttpRequest> => {
+      return new Promise(function (resolve, reject) {
```
Comment (webistomin) [inline src/nanogram.ts:54]: Can you rewrite it to an arrow function?

`js
return new Promise((resolve, reject) => {..})
`

**Comment by webistomin** [comment]
@all-contributors please add @yusufkhan07 for code, tests and bug reports

**Comment by yusufkhan07** [comment]
> Nicely done!😊 Thanks.

My pleasure :)

**Pull request merged**